brainiak/fcma/classifier.py

```diff
@@ -350,63 +377,118 @@ def fit(self, X, y, num_training_samples=None):
                     raise RuntimeError('the kernel matrix will be '
                                        'computed portion by portion, '
                                        'the test samples must be predefined '
                                        'by specifying '
                                        'num_training_samples')
-                if num_training_samples >= num_samples:
+                if num_training_samples >= self.num_samples_:
                     raise ValueError('the number of training samples '
                                      'must be smaller than '
                                      'the number of total samples')
             data, normalized_corr_data = \
-                self._compute_kernel_matrix_in_portion(X)
-            if self.num_processed_voxels >= num_voxels:
+                self._compute_kernel_matrix_in_portion(x1, x2)
+            if self.num_processed_voxels >= self.num_voxels_:
                 # training data is in shape
                 # [num_samples, num_voxels * num_voxels]
                 self.training_data_ = normalized_corr_data.reshape(
-                    num_samples,
-                    num_voxels * num_voxels)
+                    self.num_samples_,
+                    self.num_features_)
             else:
                 # do not store training data because it was partially computed
                 self.training_data_ = None
             logger.debug(
                 'kernel computation done'
             )
+        return data
+
+    def fit(self, X, y, num_training_samples=None):
+        """ use correlation data to train a model
+
+        first compute the correlation of the input data,
+        and then normalize within subject
+        if more than one sample in one subject,
+        and then fit to a model defined by self.clf.
+
+        Parameters
+        ----------
+        X: tuple (x1, x2)
+            x1 and x2 are lists of numpy array in shape [num_TRs, num_voxels],
+            they contain the activity data filtered by ROIs
+            and prepared for correlation computation.
+            Assuming within list, all elements have the same num_voxels value
+        y: labels, 1D numpy array
+            len(X) equals len(y)
+        num_training_samples: Optional[int]
+            Default None.
+            The number of samples used in the training,
+            which is set when the kernel matrix is constructed
+            portion by portion so the similarity vectors of the
+            test data have to be computed here.
+            This is ONLY set when sklearn.svm.SVC with
+            precomputed kernel is used.
+            If it is set, only those samples will be used to fit the model.
+
+        Returns
+        -------
+        self: return the object itself
+        """
+        time1 = time.time()
+        # if X is not a tuple, make it one but replicating it
+        if not isinstance(X, tuple):
+            X = (X, X)
```

Comment:
This allows users to input a single `X`. Let me explain this in the docstring instead of removing it.